src/SDK.ts

```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
+*/
+declare enum NavigationContextLevels {
+    None = 0,
+    /**
+    * Root level in Azure.
+    */
+    Deployment = 1,
+    /**
+    * Root level in on premises. Neither of {collection}, {project} and {team} tokens have information
+    */
+    Application = 2,
+    /**
+    * Flag to show {collection} token has information.
+    */
+    Collection = 4,
+    /**
+    * Flag to show {project} token has information.
+    */
+    Project = 8,
+    /**
+    * Flag to show {team} token has information.
+    */
+    Team = 16,
+    /**
+    * Sugar for all application levels.
+    */
+    ApplicationAll = 30,
+    /**
+    * Sugar for all levels
+    */
+    All = 31,
+}
+
+/**
+* Structure to specify current navigation context of the executing request. The navigation context content's are generally obtained from the request URL. Some context specifiers such as "Account" can be implicit and might come from current IVssServiceHost.
+*/
+interface NavigationContext {
+    /**
+    * A token to show which area the request has been targeted to. By default there are two areas "Admin" and "Api". They can be specified in the URL as _admin and _api respectively.
+    */
+    area: string;
+    /**
+    * Command name for the current request's route. Used in telemetry and reporting.
+    */
+    commandName: string;
+    /**
+    * Current action route value
+    */
+    currentAction: string;
+    /**
+    * Current controller route value
+    */
+    currentController: string;
+    /**
+    * Current parameters route value (the path after the controller and action in the url)
+    */
+    currentParameters: string;
+    /**
+    * The id of the matched route
+    */
+    routeId: string;
+    /**
+    * The templates for the matched route
+    */
+    routeTemplates: string[];
+    /**
+    * The set of route values for this request
+    */
+    routeValues: { [key: string]: string; };
+    /**
+    * Flag to show top most navigation context. For example the URL http://server:port/collection/project/_controller/action sets the Project bit while the URL http://server:port/collection/project/_admin/_controller/action sets also sets the area property to Admin.
+    */
+    topMostLevel: NavigationContextLevels;
+}
+
+interface DaylightSavingsAdjustmentEntry {
+    /**
+    * Millisecond adjustment from UTC
+    */
+    offset: number;
+    /**
+    * Date that the offset adjustment starts
+    */
+    start: Date;
+}
+
+interface TimeZonesConfiguration {
+    daylightSavingsAdjustments: DaylightSavingsAdjustmentEntry[];
+}
+
+interface MailSettings {
+    enabled: boolean;
+}
+
+/**
+* Web Access configuration data. This information is used to process requests on the server.  This data is also placed in a json island on each page in order for JavaScript to know key configuration data required to things like construct proper urls
+*/
+interface ConfigurationContext {
+    isHosted: boolean;

```

Comment:
Let's add a comment, something along the lines of "Distinguish between Azure DevOps Services (true) and Azure DevOps Server (false)"

Reply:
`isHosted` is present under the `IHostContext` interface.
Resolving it as content is removed. Now, there's a 1:1 reflection with the payload being returned by the SDK provider.